scripts/db_migrator.py

```diff
@@ -618,6 +618,25 @@ def migrate_telemetry(self):
         if not certs:
             self.configDB.set_entry("TELEMETRY", "certs", telemetry_data.get("certs"))
 
+    def migrate_gnmi(self):
+        if self.config_src_data:
+            if 'GNMI' in self.config_src_data:
+                # If there's GNMI in minigraph or golden config, copy configuration from config_src_data
+                gnmi_data = self.config_src_data['GNMI']
+                log.log_notice('Migrate GNMI configuration')
+                if 'gnmi' in gnmi_data:
+                    self.configDB.set_entry("GNMI", "gnmi", gnmi_data.get('gnmi'))
```

Comment:
>self.configDB [](http://example.com/codeflow?start=20&length=13)

There are upgrade/downgrade use cases. If self.configDB['GNMI'] already exists, you should not modify, right?

Reply:
I think database version is used to detect upgrade and downgrade, and maybe db_migrator will not invoke migrate_gnmi for downgrade?